src/ui/pages/Notifications/NotificationItem.tsx

```diff
@@ -26,9 +26,7 @@ const NotificationItem = ({
   const [notificationLabelText, setNotificationLabelText] =
     useState<string>("");
   const [loading, setLoading] = useState(true);
-  const connection = connectionsCache.filter(
-    (connection) => connection.id === item.connectionId
-  )[0]?.label;
+  const connection = connectionsCache?.[item.connectionId]?.label;
```

Comment:
OK, so some notifications will use a normal connection and others will use a multi-sig connection. So this line should be moved from here.
- Completely remove the call to `getMultisigIcpDetails` and remove `multiSigIcpDetails` parameter from `fetchNotificationLabel` and `notificationLabel`
- In `notificationLabel`,
  - use `connectionsCache[item.connectionId]?.label` if a credential, OR,
  - use `multisigConnectionsCache[item.connectionId]?.label` if multi-sig

I think, based on my previous comments in this PR that this will work.

^ This was the main intention for this PR - to remove so many calls to `getMultisigIcpDetails` in case we have loads of notifications for multi-sig in the list